### PR TITLE
Better errors from Atom Submission

### PR DESCRIPTION
**Pull request by talekhinezh**

### CODE REVIEW

File: radixdlt-java/src/main/java/com/radixdlt/client/core/network/epics/SubmitAtomEpic.java [hunk 1]
```diff
@@ -84,14 +84,19 @@ private Completable waitForConnection(RadixNode node) {
...
+									emitter.onNext(
+										SubmitAtomStatusAction.fromStatusNotification(
+											request.getUuid(),
+											request.getAtom(),
+											node,
+											new AtomStatusNotification(AtomStatus.EVICTED_INVALID_ATOM, submitAtomException.getData())
+										));
+									emitter.onNext(SubmitAtomCompleteAction.of(request.getUuid(), request.getAtom(), node));
+								} else {
+									emitter.onError(e);
```
Comment (laszlourszuly) [inline radixdlt-java/src/main/java/com/radixdlt/client/core/network/epics/SubmitAtomEpic.java]: From this code it seems `SubmitAtomException`'s are delivered as "success" and all other exceptions are breaking "errors". Would that expectation be clear from the context on the receiving end?

File: radixdlt-java/src/main/java/com/radixdlt/client/core/network/epics/SubmitAtomEpic.java [hunk 2]
```diff
@@ -84,14 +84,19 @@ private Completable waitForConnection(RadixNode node) {
...
-									));
-								emitter.onNext(SubmitAtomCompleteAction.of(request.getUuid(), request.getAtom(), node));
+								if (e instanceof SubmitAtomException) {
+									SubmitAtomException submitAtomException = (SubmitAtomException) e;
+									emitter.onNext(
+										SubmitAtomStatusAction.fromStatusNotification(
+											request.getUuid(),
+											request.getAtom(),
+											node,
+											new AtomStatusNotification(AtomStatus.EVICTED_INVALID_ATOM, submitAtomException.getData())
```
[the remark this answers is not in the document]
Comment (talekhinezh) [reply radixdlt-java/src/main/java/com/radixdlt/client/core/network/epics/SubmitAtomEpic.java]: Yes, there is definitely still room for improvement in really nailing the errors from here down. But yes, SubmitAtomException is thrown when an error is returned from the server indicating atom wasn't queued.

File: radixdlt-java/src/main/java/com/radixdlt/client/core/network/epics/SubmitAtomEpic.java (hunk 1 above)
Comment (talekhinezh) [reply radixdlt-java/src/main/java/com/radixdlt/client/core/network/epics/SubmitAtomEpic.java]: Yes, this is seen in `RadixApplicationAPI.java:105` + `RadixApplicationAPI.java:946`

File: radixdlt-java/src/main/java/com/radixdlt/client/core/network/epics/SubmitAtomEpic.java [hunk 3]
```diff
@@ -84,14 +84,19 @@ private Completable waitForConnection(RadixNode node) {
...
-										request.getAtom(),
-										node,
-										new AtomStatusNotification(AtomStatus.EVICTED_INVALID_ATOM, new JsonObject())
-									));
-								emitter.onNext(SubmitAtomCompleteAction.of(request.getUuid(), request.getAtom(), node));
+								if (e instanceof SubmitAtomException) {
+									SubmitAtomException submitAtomException = (SubmitAtomException) e;
+									emitter.onNext(
+										SubmitAtomStatusAction.fromStatusNotification(
+											request.getUuid(),
```
Comment (mpsandiford) [inline radixdlt-java/src/main/java/com/radixdlt/client/core/network/epics/SubmitAtomEpic.java]: Probably need to look at refactoring this into smaller units of code at some point.  Have reached 11 levels of indent here, which is a bit much really.

**Pull request merged**